File: files/300-399/318/CH-318.py
import pandas as pd
import numpy as np

path = "300-399/318/CH-318 Table Transformation.xlsx"
input = pd.read_excel(path, usecols="B:E", skiprows=1, nrows=9, header=None).to_numpy()
test = pd.read_excel(path, usecols="G:J", skiprows=1, nrows=5).rename(columns=lambda c: c.replace('.1', ''))  

m = np.where(np.char.startswith(input.astype(str), 'Column'), np.nan, input)

for j in range(m.shape[1]):
    col = m[:, j][~pd.isna(m[:, j])]
    m[:len(col), j] = col
    m[len(col):, j] = np.nan

result = pd.DataFrame(m, columns=test.columns)
result = result.iloc[1:][~pd.isna(result['Date'])].reset_index(drop=True)
if 'Date' in result.columns:
    result['Date'] = pd.to_datetime(result['Date']).dt.strftime('%Y-%m-%d %H:%M')

print((result == test).all().all())

File: files/300-399/318/CH-318.R
library(tidyverse)
library(readxl)
library(janitor)

path = "files/300-399/318/CH-318 Table Transformation.xlsx"
input = read_excel(path, range = "B2:E10", col_names = FALSE)
test  = read_excel(path, range = "G2:J7")

m = as.matrix(input)
m[str_starts(m, "Column")] = NA

df = m %>%
    as_tibble() %>%
    map_df(~ {
        x = na.omit(.x)
        c(x, rep(NA, nrow(m) - length(x)))
    }) %>%
    row_to_names(1) %>%
    drop_na() %>%
    mutate(Date = excel_numeric_to_date(as.numeric(Date)) %>% as.POSIXct(),
        Quantity = as.numeric(Quantity))

all.equal(result, test)
